pkgs/applications/networking/xpipe/default.nix

```
@@ -0,0 +1,66 @@
+{ stdenv, lib, fetchzip, makeDesktopItem }:
+
+let
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
+  arch = {
+    x86_64-linux = "x86_64";
+    aarch64-linux = "arm64";
+  }.${system} or throwSystem;
+
+  sha256 = {
+    x86_64-linux = "VmJa0DqnFSzq4wJ+HchguDK4tMeKP/XaE2MidXfootM=";
+    aarch64-linux = "LdXfhZK4n8HqN1u/rKIIZhFk7fREXbTuAzyvvhFR6Lo=";
+  }.${system} or throwSystem;
```

Both of these seem to only be used to fetch the source. Maybe we could just fetch both sources here, rather than split up the parts?

```suggestion
  version = "1.6.0";
  src = let 
    inherit (stdenv.hostPlatform) system;
  in {
    x86_64-linux = fetchzip {
      url = "https://github.com/xpipe-io/xpipe/releases/download/${version}/xpipe-portable-linux-x86_64.tar.gz";
      hash = "sha256-VmJa0DqnFSzq4wJ+HchguDK4tMeKP/XaE2MidXfootM=";
    };
    aarch64-linux = fetchzip {
      url = "https://github.com/xpipe-io/xpipe/releases/download/${version}/xpipe-portable-linux-arm64.tar.gz";
      hash = "sha256-LdXfhZK4n8HqN1u/rKIIZhFk7fREXbTuAzyvvhFR6Lo=";
    };
  }.${system} or throw "Unsupported system: ${system}";
```

Then we could just do `inherit version src` in the `mkDerivation`;

---

Yeah right now they are just used for fetching the source but I think it is good to have the arch variables there for the future as I might have to add some architecture specific parts as both versions are not the same, mainly due to the x86-64 version using musl and the arm version not using it.

---

Well, if for a reason it's probably fine. However, I think the hashes should be changed to SRI format

```nix
  hash = {
    x86_64-linux = "sha256-VmJa0DqnFSzq4wJ+HchguDK4tMeKP/XaE2MidXfootM=";
    aarch64-linux = "sha256-LdXfhZK4n8HqN1u/rKIIZhFk7fREXbTuAzyvvhFR6Lo=";
  }.${system} or throwSystem;
```